drivers/power/ltc7841/ltc7841.c

```diff
@@ -0,0 +1,435 @@
+	config = register_data[0];
+	config &= LTC7841_NOMINAL_SLEW_RATE;
+
+	return ltc7841_reg_write(desc, LTC7841_MFR_CONFIG, config);
+}
+/**
+ * @brief Change to Slow Slew Rate for LTC7841
+ * @param desc - The device structure.
+ * @return 0 in case of success, error code otherwise
+ */
+int ltc7841_change_to_slow_slew_rate(struct ltc7841_desc *desc)
+{
+	int ret;
+	uint8_t config;
+	uint8_t register_data[LTC7841_BYTE_LENGTH];
+
+	ret = ltc7841_reg_read(desc, LTC7841_MFR_CONFIG, register_data);
+	if (ret)
+		return ret;
+	/* get latest data for MFR_CONFIG */
+	config = register_data[0];
+	config |= LTC7841_SLOW_SLEW_RATE;
+
+	return ltc7841_reg_write(desc, LTC7841_MFR_CONFIG, config);
+}
+/**
+ * @brief Change to Fast Slew Rate for LTC7841
+ * @param desc - The device structure.
+ * @return 0 in case of success, error code otherwise
+ */
+int ltc7841_change_to_fast_slew_rate(struct ltc7841_desc *desc)
+{
+	int ret;
+	uint8_t config;
+	uint8_t register_data[LTC7841_BYTE_LENGTH];
+
+	ret = ltc7841_reg_read(desc, LTC7841_MFR_CONFIG, register_data);
+	if (ret)
+		return ret;
+	/* get latest data for MFR_CONFIG */
+	config = register_data[0];
+	config |= LTC7841_FAST_SLEW_RATE;
+
+	return ltc7841_reg_write(desc, LTC7841_MFR_CONFIG, config);
+}
+/**
+ * @brief Change to Internal Die Temeperature Reading for LTC7841
+ * @param desc - The device structure.
+ * @return 0 in case of success, error code otherwise
+ */
+int ltc7841_change_to_internal_die_temperature(struct ltc7841_desc *desc)
+{
+	int ret;
+	uint8_t config;
+	uint8_t register_data[LTC7841_BYTE_LENGTH];
+
+	ret = ltc7841_reg_read(desc, LTC7841_MFR_CONFIG, register_data);
+	if (ret)
+		return ret;
+	/* get latest data for MFR_CONFIG */
+	config = register_data[0];
+	config &= ((LTC7841_INTERNAL_DIE_TEMPERATURE) << READ_TEMPERATURE_1_BIT);
+	return ltc7841_reg_write(desc, LTC7841_MFR_CONFIG, config);
+}
+/**
+ * @brief Change to TSNS PIN VOLTAGE for LTC7841
+ * @param desc - The device structure.
+ * @return 0 in case of success, error code otherwise
+ */
+int ltc7841_change_to_tsns_pin_voltage(struct ltc7841_desc *desc)
+{
+	int ret;
+	uint8_t config;
+	uint8_t register_data[LTC7841_BYTE_LENGTH];
+
+	ret = ltc7841_reg_read(desc, LTC7841_MFR_CONFIG, register_data);
+	if (ret)
+		return ret;
+	/* get latest data for MFR_CONFIG */
+	config = register_data[0];
+	config |= ((LTC7841_TSNS_PIN_VOLTAGE) << READ_TEMPERATURE_1_BIT);
+	return ltc7841_reg_write(desc, LTC7841_MFR_CONFIG, config);
+}
+/**
+ * @brief Change Vout command
+ * @param desc - The device structure.
+ * @param vout_level - value is from 0 to 0x1FF
+ * @return 0 in case of success, error code otherwise
+ */
+int ltc7841_change_vout_command(struct ltc7841_desc *desc, uint16_t vout_level)
+{
+	int ret;
+
+	if (vout_level <= MFR_MAX_VALUE) {
+		/* okay to proceed */
+		ret = ltc7841_reg_write(desc, LTC7841_MFR_VOUT_COMMAND, vout_level);
+		if (ret)
+			return ret;
+	} else {
+		ret = -EINVAL;
+	}
+	return ret;
```

Comment:
another style remark, it's kind of informal but we always put a blank line before the final return statement of the function like:
```
    }

    return ret;
}
```